src/web/src/components/SearchProvider.tsx

```diff
@@ -47,47 +72,86 @@ const SearchProvider = ({ children }: Props) => {
   const textParam = Array.isArray(router.query.text)
     ? router.query.text[0]
     : router.query.text || '';
-  const filterParam = router.query.filter === 'post' || !router.query.filter ? 'post' : 'author';
+  const filterParam = getFilter(router.query.filter);
 
   // We manage the state of `text` and `filter` internally, and update URL on
   // form submit only.  These are used in the <SearchBar>, and the user can change them.
-  const [text, setText] = useState('');
-  const [filter, setFilter] = useState<FilterProp['filter']>('post');
+  const [post, setPost] = useState<string>('');
+  const [author, setAuthor] = useState<string>('');
+  const [title, setTitle] = useState<string>('');
+  const [from, setFrom] = useState<string>('');
+  const [to, setTo] = useState<string>('');
+
   const [showHelp, setShowHelp] = useState(true);
 
+  // const buildDateQuery = () => {
+  //   let query = '/search/?';
+  //   if (to) {
+  //     query += `post`;
+  //   }
+  // };
+
   const onSubmitHandler = (event: FormEvent) => {
     event.preventDefault();
-    router.push(`/search?text=${text}&filter=${filter}`);
+    // if (to || from) {
+    //   const query = buildDateQuery();
+    // }
+    router.push(`/search?/post=${post}&author=${author}&title=${title}`);
   };
 
   const toggleHelp = (value: boolean) => {
     setShowHelp(value);
   };
 
-  const onTextChange = (value: string) => {
-    setText(value);
+  const onPostChange = (value: string) => {
+    setPost(value);
+  };
+
+  const onTitleChange = (value: string) => {
+    setTitle(value);
+  };
+
+  const onAuthorChange = (value: string) => {
+    setAuthor(value);
+  };
+
+  const onStartDateChange = (value: string) => {
+    setFrom(value);
   };
 
-  const onFilterChange = (value: FilterProp['filter']) => {
-    setFilter(value);
+  const onEndDateChange = (value: string) => {
+    console.log('end date', value);
+    if (value) {
+      setTo(value);
+    } else {
+      setTo(new Date().toISOString().split('T')[0]);
+    }
   };
 
   useEffect(() => {
-    setText(textParam);
-    setFilter(filterParam);
+    setPost(textParam);
+    setAuthor(textParam);
+    setFrom(textParam);
+    setTo(textParam);
+    setTitle(textParam);
   }, [textParam, filterParam]);
 
   return (
     <SearchContext.Provider
       value={{
-        text,
-        textParam,
```

Comment:
Sorta related to the SearchResults bug, `textParam` is different from `text` in the current iteration which is what keeps the state from being updated constantly. Might want to incorporate something similar here